common/models/acl.js

```diff
@@ -204,11 +205,12 @@ module.exports = function(ACL) {
   /*!
    * Resolve permission from the ACLs
    * @param {Object[]) acls The list of ACLs
-   * @param {Object} req The request
-   * @returns {AccessRequest} result The effective ACL
+   * @param {AccessRequest} req The access request
+   * @returns {AccessRequest} result The resolved access request
    */
   ACL.resolvePermission = function resolvePermission(acls, req) {
     if (!(req instanceof AccessRequest)) {
+      req.registry = this.registry;
       req = new AccessRequest(req);
```

Comment:
hi @ejvaitkus : i'm afraid it's not specific to the `authorizedRoles` property as this remote method argument is also used to pass remoting context properties downstream to hooks and methods. see  https://loopback.io/doc/en/lb3/Using-current-context.html

it's on purpose that the generic arg `options` is used as is in LB3.